### PR TITLE
Fix a conhost binary size regression due to fmt

### PR DESCRIPTION
**Pull request by lhecker**
6140fd9 causes a binary size regression in conhost.
This PR fixes most if not all of the regression, by replacing `FMT_STRING`
with `FMT_COMPILE` allowing us to drop most of the formatters built
into fmt during linking (for instance floating point formatters).

Additionally `std::wstring` was replaced with `fmt::basic_memory_buffer`
in the same vein as was done for VtEngine. Stack is
cheap and this prevents any unnecessary allocations.

## PR Checklist
* [x] I work here
* [x] Tests added/passed

## Validation Steps Performed
* vttest 11.2.5.3.6.7 and .8 (DECSTBM and SGR) complete successfully ✅

### CODE REVIEW

File: src/terminal/adapter/adaptDispatch.cpp [hunk 1]
```diff
@@ -2509,62 +2509,64 @@ ITermDispatch::StringHandler AdaptDispatch::RequestSetting()
 // - None
 void AdaptDispatch::_ReportSGRSetting() const
 {
+    using namespace std::string_view_literals;
+
     // A valid response always starts with DCS 1 $ r.
     // Then the '0' parameter is to reset the SGR attributes to the defaults.
-    std::wstring response = L"\033P1$r0";
+    fmt::basic_memory_buffer<wchar_t, 64> response;
```
Comment (DHowett) [inline src/terminal/adapter/adaptDispatch.cpp:2516]: is 64 long enough?

**Review by DHowett — APPROVED** [review]
nice!

File: src/terminal/adapter/adaptDispatch.cpp (hunk 1 above)
Comment (lhecker) [reply src/terminal/adapter/adaptDispatch.cpp:2516]: Yeah I was counting with each number being 2 digits and all parameters present and I got 56 chars.
For comparison: `std::wstring` comes equipped with a whopping 7 chars for the SSO buffer.

**Review by zadjii-msft — APPROVED** [review]

**Comment by DHowett** [comment]
@msftbot merge this in 8 minutes

**Pull request merged**